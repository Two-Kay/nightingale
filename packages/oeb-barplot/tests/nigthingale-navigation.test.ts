import NightingaleNavigation from "../src/index";

let rendered: NightingaleNavigation;

describe("nightingale-navigation tests", () => {
  beforeEach(() => {
    rendered = new NightingaleNavigation();
    rendered.setAttribute("height", "15");
    document.documentElement.appendChild(rendered);
  });

  afterEach(() => {
    document.documentElement.removeChild(rendered);
  });

  test("it should display the navigation correctly", async () => {
    expect(rendered).toMatchSnapshot();
  });

  test("it should display the navigation correctly after zoom", () => {
    rendered.setAttribute("display-start", "2");
    rendered.setAttribute("display-end", "4");
    return new Promise((done) => {
      window.requestAnimationFrame(() => {
        expect(rendered).toMatchSnapshot();
        done(true);
      });
    });
  });
});
