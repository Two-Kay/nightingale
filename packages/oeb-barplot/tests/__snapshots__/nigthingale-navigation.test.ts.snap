// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nightingale-navigation tests it should display the navigation correctly 1`] = `
<oeb-barplot
  height="15"
>
  <!---->
  <div
    class="container"
    style="padding: 0px 10px 0px 10px;"
  >
    <svg
      height="15"
      id=""
      width="800"
    >
      <text
        class="start-label"
        x="0"
        y="5"
      >
        1
      </text>
      <text
        class="end-label"
        text-anchor="end"
        x="800"
        y="5"
      >
        1
      </text>
      <g
        class="x axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
      >
        <path
          class="domain"
          d="M10.5,6V0.5H790.5V6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(10.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            1.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(88.49999999999999,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.9
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(166.49999999999997,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.8
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(244.50000000000003,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.7
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(322.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.6
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(400.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(478.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.4
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(556.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.3
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(634.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.2
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(712.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.1
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(790.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.0
          </text>
        </g>
      </g>
      <g
        class="brush"
        fill="none"
        pointer-events="all"
      >
        <rect
          class="overlay"
          cursor="crosshair"
          height="7.65"
          pointer-events="all"
          width="780"
          x="10"
          y="0"
        />
        <rect
          class="selection"
          cursor="move"
          fill="#777"
          fill-opacity="0.3"
          height="7.65"
          shape-rendering="crispEdges"
          stroke="#fff"
          style=""
          width="0"
          x="10"
          y="0"
        />
        <rect
          class="handle handle--w"
          cursor="ew-resize"
          height="13.65"
          style=""
          width="6"
          x="7"
          y="-3"
        />
        <rect
          class="handle handle--e"
          cursor="ew-resize"
          height="13.65"
          style=""
          width="6"
          x="7"
          y="-3"
        />
      </g>
      <polygon
        class="zoom-polygon"
        fill="#777"
        fill-opacity="0.3"
        points="10,7.5
        10,7.5
        800,15
        0,15"
      />
    </svg>
  </div>
</oeb-barplot>
`;

exports[`nightingale-navigation tests it should display the navigation correctly after zoom 1`] = `
<oeb-barplot
  display-end="4"
  display-start="2"
  height="15"
>
  <!---->
  <div
    class="container"
    style="padding: 0px 10px 0px 10px;"
  >
    <svg
      height="15"
      id=""
      width="800"
    >
      <text
        class="start-label"
        x="0"
        y="5"
      >
        2
      </text>
      <text
        class="end-label"
        text-anchor="end"
        x="800"
        y="5"
      >
        4
      </text>
      <g
        class="x axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
      >
        <path
          class="domain"
          d="M10.5,6V0.5H790.5V6"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(10.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            1.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(88.49999999999999,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.9
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(166.49999999999997,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.8
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(244.50000000000003,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.7
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(322.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.6
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(400.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(478.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.4
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(556.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.3
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(634.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.2
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(712.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.1
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(790.5,0)"
        >
          <line
            stroke="currentColor"
            y2="6"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="9"
          >
            0.0
          </text>
        </g>
      </g>
      <g
        class="brush"
        fill="none"
        pointer-events="all"
      >
        <rect
          class="overlay"
          cursor="crosshair"
          height="7.65"
          pointer-events="all"
          width="780"
          x="10"
          y="0"
        />
        <rect
          class="selection"
          cursor="move"
          fill="#777"
          fill-opacity="0.3"
          height="7.65"
          shape-rendering="crispEdges"
          stroke="#fff"
          style=""
          width="-1560"
          x="-770"
          y="0"
        />
        <rect
          class="handle handle--w"
          cursor="ew-resize"
          height="13.65"
          style=""
          width="6"
          x="-773"
          y="-3"
        />
        <rect
          class="handle handle--e"
          cursor="ew-resize"
          height="13.65"
          style=""
          width="6"
          x="-2333"
          y="-3"
        />
      </g>
      <polygon
        class="zoom-polygon"
        fill="#777"
        fill-opacity="0.3"
        points="-770,7.5
        -2330,7.5
        800,15
        0,15"
      />
    </svg>
  </div>
</oeb-barplot>
`;
